great_expectations/experimental/datasources/spark_datasource.py

```diff
@@ -0,0 +1,229 @@
+from __future__ import annotations
+
+import copy
+import logging
+import pathlib
+import re
+from typing import TYPE_CHECKING, Dict, List, Optional, Pattern, Tuple, Type, Union
+
+import pydantic
+from typing_extensions import ClassVar, Literal
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.alias_types import PathStr
+from great_expectations.core.batch_spec import PathBatchSpec
+from great_expectations.experimental.datasources.interfaces import (
+    Batch,
+    BatchRequest,
+    BatchRequestOptions,
+    BatchSortersDefinition,
+    DataAsset,
+    Datasource,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.execution_engine import ExecutionEngine
+
+LOGGER = logging.getLogger(__name__)
+
```

Comment:
Nit - why do we use `LOGGER` some places and `logger` elsewhere? Can we standardize?

Reply:
@cdkini I am not sure, but in ZEP the style has been capitalized "LOGGER" -- perhaps @Kilo59 and/or @billdirks could provide more background.  Thank you.

Reply:
I'm fine with changing it.
This is just an old `pylint` habit. Pylint wants any model level constants to be `UPPER_CASE`.
>`const` - Module-level constants, any variable defined at module level that is not bound to a class object.
https://pylint.pycqa.org/en/latest/user_guide/messages/convention/invalid-name.html#invalid-name-c0103

But we don't use `pylint` so... 😄

Reply:
@Kilo59 @cdkini @billdirks -- I am happy to convert all of our `logger.` usage to `LOGGER.` in an immediately following pull request -- after the present pull request is merged with `logger.` spelled as in the working branch now.  Thanks.

Reply:
That seems like unnecessary churn that's going to lead to conflicts. IMO it would be bett to drop/revert the changes to `LOGGER`.

Reply:
@Kilo59 @billdirks @cdkini While we previously had a motion to replace `logger` (all lowercase) with `LOGGER` (all uppercase) and even though doing so has not caused any checks to fail, not everybody is on-board with this change at the present time.  Hence, per the recommendation by @Kilo59 I am reverting it (we can always apply it at a later date, should we rationalize that this aspect of adhering to the Google Python style is beneficial to GX OSS.  Thank you. /cc @donaldheppner @5u2ie